### ISSUE TITLE
Investigate and fix network calls on wrong Threads

### ISSUE THREAD

**Issue by abrooksv** [aws/aws-toolkit-jetbrains#865]
After fixing our assert statement, it has become evident we managed to put a few calls on the UI thread. We need to move these calls to the background / async.

#863 fixed some performance issues on start up, but it became evident that our credential switching / validation logic is wrong since it relies on UI calls and can't work in an async manor. We need to investigate the switching UX and implement it that is async friendly.

**Comment by abrooksv** [comment]
Changes to credential flow:

1. User changes credentials
2. Toolkit broadcasts out a new state change across the `ACCOUNT_SETTINGS_CHANGED` MessageBus. This state represents that credentials are loading
3. Explorer and status bar go into a Loading Credentials state
4a. If credentials load, `ProjectAccountSettingsManager#activeCredentialProvider` will be set to the requested credentials
4b. If credentials fail,  `ProjectAccountSettingsManager#activeCredentialProvider` is nulled out
5. New state change is broadcasted with the new value

**Comment by abrooksv** [comment]
Credential switching has been fixed:

Remaining issue(s) is:
1. expanding child nodes for CFN node
```
	at software.amazon.awssdk.services.cloudformation.DefaultCloudFormationClient.describeStackResources(DefaultCloudFormationClient.java:1195)
	at software.amazon.awssdk.services.cloudformation.CloudFormationClient.describeStackResources(CloudFormationClient.java:1938)
	at software.aws.toolkits.jetbrains.services.cloudformation.CloudFormationStackNode.loadServerlessStackResources(CloudFormationExplorerNodes.kt:119)
	at software.aws.toolkits.jetbrains.services.cloudformation.CloudFormationStackNode.updateCachedChildren(CloudFormationExplorerNodes.kt:105)
	at software.aws.toolkits.jetbrains.services.cloudformation.CloudFormationStackNode.getChildren(CloudFormationExplorerNodes.kt:90)
	at software.aws.toolkits.jetbrains.core.explorer.ExplorerToolWindow$createTree$2.treeWillExpand(ExplorerToolWindow.kt:125)
	at javax.swing.JTree.fireTreeWillExpand(JTree.java:2839)
	at javax.swing.JTree.setExpandedState(JTree.java:3630)
	at javax.swing.JTree.expandPath(JTree.java:2219)
	at javax.swing.plaf.basic.BasicTreeUI.toggleExpandState(BasicTreeUI.java:2286)
```

**Comment by abrooksv** [comment]
Found another call:

```
	at software.amazon.awssdk.services.lambda.DefaultLambdaClient.listFunctions(DefaultLambdaClient.java:1531)
	at software.amazon.awssdk.services.lambda.paginators.ListFunctionsIterable$ListFunctionsResponseFetcher.nextPage(ListFunctionsIterable.java:128)
	at software.amazon.awssdk.services.lambda.paginators.ListFunctionsIterable$ListFunctionsResponseFetcher.nextPage(ListFunctionsIterable.java:119)
	at software.amazon.awssdk.core.pagination.sync.PaginatedResponsesIterator.next(PaginatedResponsesIterator.java:58)
	at software.amazon.awssdk.core.pagination.sync.PaginatedItemsIterable$ItemsIterator.<init>(PaginatedItemsIterable.java:58)
	at software.amazon.awssdk.core.pagination.sync.PaginatedItemsIterable.iterator(PaginatedItemsIterable.java:48)
	at software.aws.toolkits.jetbrains.core.DefaultAwsResourceCache$lambdaFunctions$1.apply(AwsResourceCache.kt:58)
	at software.aws.toolkits.jetbrains.core.DefaultAwsResourceCache$lambdaFunctions$1.apply(AwsResourceCache.kt:26)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at software.aws.toolkits.jetbrains.core.DefaultAwsResourceCache.lambdaFunctions(AwsResourceCache.kt:44)
	at software.aws.toolkits.jetbrains.services.lambda.upload.LambdaLineMarker.handlerInRemote(LambdaLineMarker.kt:96)
	at software.aws.toolkits.jetbrains.services.lambda.upload.LambdaLineMarker.shouldShowLineMarker(LambdaLineMarker.kt:85)
	at software.aws.toolkits.jetbrains.services.lambda.upload.LambdaLineMarker.getLineMarkerInfo(LambdaLineMarker.kt:48)
	at com.intellij.codeInsight.daemon.impl.LineMarkersPass.queryProviders(LineMarkersPass.java:171)
```

**Comment by zhangzhx** [comment]
This fix has been released in [version 1.3](https://plugins.jetbrains.com/plugin/11349-aws-toolkit/update/61424)
